### PR TITLE
WIP: Add README docs

### PR DESCRIPTION
**Pull request by jxnblk**
- Rendered version: https://github.com/gatsbyjs/gatsby-starter-landing-page/blob/update-readme/README.md
- Theme README: https://github.com/gatsbyjs/gatsby-starter-landing-page/blob/update-readme/gatsby-theme-landing-page/README.md

Other unrelated changes:

- Added exports to `gatsby-theme-landing-page/index.js` for MarkdownText and Link; I think this is a good way to expose a more public API for helper components/hooks/etc that someone could use when shadowing or extending the theme. I didn't add everything to this, but put a couple in as as example
- Added `.gitkeep` to the site's `src/gatsby-theme-landing-page` shadow directory to make it easier and more foolproof to shadow components in the theme; we should consider what sort of placeholder shadow files we want to include in the final starter

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,223 @@
...
+  /* shadows */
+  --box-shadow-raised: 0px 1px 2px rgba(46, 41, 51, 0.08), 0px 2px 4px rgba(71, 63, 79, 0.08);
+  --box-shadow-floating: 0px 2px 4px rgba(46, 41, 51, 0.08), 0px 4px 8px rgba(71, 63, 79, 0.16);
+}
+```
+
+## Customizing and extending components
+
+To customize a landing page section component, create a file in `src/gatsby-theme-landing-page/sections/` with the same name used in the theme. This will shadow the built-in component to completely override it.
+
```
Comment (thinkybeast) [inline README.md:192]: Might be helpful to include a screenshot here of what we're talking about, as it took me a moment to understand which part of Contentful this is referring to

File: README.md [hunk 2]
```diff
@@ -1,3 +1,223 @@
...
+To add more components and extend the functionality, create a `src/gatsby-theme-landing-page/sections/index.js` file that exports the named components that you'd like to use. Be sure to update your Contentful space's content model to reflect these changes by ensuring the LandingPageSection's _Component_ field validation includes all possible options.
```
Comment (thinkybeast) [inline README.md:192]: Might be helpful to include a screenshot here of what part of Contentful we're talking about, as it took me a moment to understand which page this was referring to

File: gatsby-theme-landing-page/README.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+# .env
+CONTENTFUL_SPACE_ID="<SPACE_ID>"
+CONTENTFUL_DELIVERY_ACCESS_TOKEN="<ACCESS_TOKEN>"
+```
+
+<!-- TODO: update these instructions once the JSON file is created -->
+
+Upload the `data/sample-data.json` to your Contentful space to set up the content model. Then, create your first landing page in Contentful.
+
+## Customizing typography, colors, and layout
```
Comment (thinkybeast) [inline gatsby-theme-landing-page/README.md:45]: Can we remove these?

**Review by thinkybeast — APPROVED** [review]
Just a couple of non-blocking comments. Also wondering if we should give a high-level overview of the content model (and the mental model behind it) in this README

File: gatsby-theme-landing-page/README.md (hunk 3 above)
Comment (jxnblk) [reply gatsby-theme-landing-page/README.md:45]: Yup, added some more detailed docs to the theme's README; we might want to do a review of the overall content and where each piece should live: https://github.com/gatsbyjs/gatsby-starter-landing-page/blob/update-readme/gatsby-theme-landing-page/README.md#customizing-typography-colors-and-layout

File: README.md (hunk 2 above)
Comment (jxnblk) [reply README.md:192]: I thought I left a TODO comment somewhere, but 100% agree

File: gatsby-theme-landing-page/README.md (hunk 3 above)
Comment (jxnblk) [reply gatsby-theme-landing-page/README.md:45]: Oh, I might've misunderstood your comment – I left the headings in as placeholders to remember what other details I wanted to add in, if that's what this comment was referring to

Comment (thinkybeast) [reply gatsby-theme-landing-page/README.md:45]: yup, that's all :)

**Review by thinkybeast — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,3 +1,266 @@
...
+
+   ```sh
+   npx gatsby new my-landing-page-site https://github.com/gatsbyjs/gatsby-starter-landing-page
+   ```
+
+2. **Configure your Contentful space**
+
+  <!-- TODO: update these instructions once the JSON file is created -->
+
+Create a new Contentful space or use an existing one, then upload the `data/sample-data.json` file to your space.
```
Comment (jxnblk) [inline README.md]:
```suggestion
    Create a new Contentful space or use an existing one, then upload the `data/sample-data.json` file to your space.
```

File: README.md [hunk 5]
```diff
@@ -1,3 +1,266 @@
...
+
+   ```sh
+   npx gatsby new my-landing-page-site https://github.com/gatsbyjs/gatsby-starter-landing-page
+   ```
+
+2. **Configure your Contentful space**
+
+  <!-- TODO: update these instructions once the JSON file is created -->
+
+    Create a new Contentful space or use an existing one, then upload the `data/sample-data.json` file to your space.
```
Comment (jxnblk) [inline README.md]:
```suggestion
  Create a new Contentful space or use an existing one, then upload the `data/sample-data.json` file to your space.
```

File: gatsby-theme-landing-page/README.md [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+  --max-width: 1024px;
+  --max-width-narrow: 768px;
+  --border-radius: 4px;
+  /* shadows */
+  --box-shadow-raised: 0px 1px 2px rgba(46, 41, 51, 0.08), 0px 2px 4px rgba(71, 63, 79, 0.08);
+  --box-shadow-floating: 0px 2px 4px rgba(46, 41, 51, 0.08), 0px 4px 8px rgba(71, 63, 79, 0.16);
+}
+```
+
+**Note:** when shadowing files, you may need to clean cache and restart your Gatsby development server to see new shadow files.
```
Comment (jxnblk) [inline gatsby-theme-landing-page/README.md]:
```suggestion
**Note:** when shadowing files, you may need to clear cache by running `gatsby clean` and restart your Gatsby development server to see new shadow files.
```

**Pull request merged**